### PR TITLE
pydeck: jupyter widget transport

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Background
As part of the cross-language roadmap for deck.gl we need to have a common transport API across languages. This is a first stab at refactoring the Jupyter Widget connection into a portabl API

#### Change List
- Adds base transport class to `deck.gl/json` module
- Adds jupyter transport subclass to `jupyter-widget` module
- Refactors the pydeck application into a `playground` folder - intention is to move this into a separate module and unify with all other playground examples (SwiftDeck, JavaDeck etc).

Now renders in Notebook, but to_html() might be broken @ajduberstein 

Screen Shot 2020-05-15 at 4 23 29 PM

### CODE REVIEW

**Comment by ibgreen** [comment]
@kylebarron

**Comment by ibgreen** [comment]
@ajduberstein @kylebarron 

Did a cleanup round of this, this is getting into better shape.

I am getting a weird require error in the tests - I think it is related to Es5/Es6 sublcassing issues and the way code is being built here with non-ES6 imports, I haven't figured it out yet and am not familiar with the setup so we might need to pair up to close this.

File: bindings/pydeck/examples/01 - Introduction.ipynb [hunk 1]
```diff
@@ -21,9 +21,17 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 11,
```
Comment (ajduberstein) [inline bindings/pydeck/examples/01 - Introduction.ipynb:24]: I don't yet grok dealing with Jupyter Notebooks in version control. When I commit these, I try to clear out the output cells before I do. 

We agree that it feels like the task for a linter.

**Comment by ajduberstein** [comment]
Please rebase to include the linter changes that are causing CI to fail.

**Comment by ajduberstein** [comment]
image
Works in `to_html` without an issue and works in JupyterLab, but you do need to make one additional change, `modules/jupyter-widget/src/index.js` should export `JupyterTransportModel` and `JupyterTransportView`.

Otherwise we're good to go. #4610 and #4573 will make changes like this easier to test in multiple environments in the future.

**Pull request merged**